src/Sylius/Bundle/ArchetypeBundle/Form/Type/ArchetypeType.php

```diff
@@ -44,6 +45,8 @@ function __construct($dataClass, array $validationGroups, $subject)
      */
     public function buildForm(FormBuilderInterface $builder, array $options)
     {
+        $currentArchetype = $builder->getForm()->getData();
```

Comment:
We should do a listener, cause it only works when you pass the data when creating the form. If someone calls ``setData`` later, it will not be taken into account. :)